test/functional/api/v2/template_controller_test.rb

```diff
@@ -24,6 +46,128 @@ class TemplateControllerTest < ::ActionController::TestCase
           refute Dir.entries("#{tmpdir}/provisioning_templates/provision").include?('export_test_template.erb')
         end
       end
+
+      test "should import filtered templates" do
+        post :import, params: { "repo" => "#{ForemanTemplates::Engine.root}/test/templates/core", "filter" => 'filter1', "prefix" => '' }
+        assert_response :success
+        templates = JSON.parse(@response.body)['message']['templates']
+        templates.each do |template|
+          if template['name'] == 'filter1 template'
+            assert template['imported']
+          end
+        end
+      end
+
+      test "should not import negate filtered templates" do
+        post :import, params: { "repo" => "#{ForemanTemplates::Engine.root}/test/templates/core", "filter" => 'filter1', "prefix" => '', "negate" => true }
+        assert_response :success
+        templates = JSON.parse(@response.body)['message']['templates']
+        templates.each do |template|
+          if template['name'] == 'filter1 template'
+            assert_not template['imported']
+          end
+        end
+      end
+
+      test "should import from git branch" do
+        post :import, params: { "repo" => 'https://github.com/SatelliteQE/foreman_templates.git', "prefix" => '', "branch" => "example", "associate" => "new" }
+        assert_response :success
+        templates = JSON.parse(@response.body)['message']['templates']
+        assert_equal templates.length, 11
+        templates.each do |template|
+          assert template['imported']
+        end
+      end
+
+      test "should import from git repository directory" do
+        post :import, params: { "repo" => 'https://github.com/SatelliteQE/foreman_templates.git', "prefix" => '', "branch" => "example", "associate" => "new", "dirname" => "import" }
+        assert_response :success
+        templates = JSON.parse(@response.body)['message']['templates']
+        assert_equal templates.length, 10
+        templates.each do |template|
+          assert template['imported']
+        end
+      end
+
+      test "should not update locked template on import" do
+        FactoryBot.create(:import, :repo => "#{ForemanTemplates::Engine.root}/test/templates/core", :filter => "filter1", :prefix => '', :associate => "new", :lock => true)
```

Comment:
Test are failing on this, maybe it should be `post :import` ?

Reply:
Then there will be two posts in the same test for which we are advised to not to do so. What do u think ?

Reply:
Also, I thought its failing for my system only but fortunately on ur s as well.

Reply:
There is no factory for `:import`, if you do not want to use a second `post`, then you will need to read the template files and create the templates that you need.

Reply:
I think, better I will use two posts.